### ISSUE TITLE
Hazard meta data

### ISSUE THREAD

**Issue by johcarter** [OasisLMF/ODS_OpenResultsData#9]
We've been discussing adding some ORD reports for hazard, such as hazard EP curves and hazard value by location and event.  However there is no clear structure for hazard meta data in ORD. Here is a table of the current relevant fields in ORD:

![image](https://user-images.githubusercontent.com/11421733/173828833-69c15e7e-462b-4393-b1f2-0c6cda3ab921.png)

If we support reports which contain values such as windspeeds and flood depths, we should have clearly structured meta data which references exactly what those values mean in terms of hazard intensity measure and units.  

Peril codes are used as a proxy for a hazard intensity measure, but in practice this is not well specified in model settings.

e.g WTC means Tropical Cyclone, but is normally used to represented the hazard of wind speed. This is a 'peril type' or event type which is being used as a proxy for wind speed, but they are different concepts.

A meta data structure might look like this, for example;
![image](https://user-images.githubusercontent.com/11421733/173834101-5aec3298-de76-4e0c-9360-33a8e41233c1.png)

We could also consider codifying common hazard intensity measures in ODS, e.g.

HazardIntensityMeasureType =1
name = 'windspeed'
desc = '3 second maximum peak gust speed at 10 metre elevation'
unit = 'metres per second'

We can't expect to codify all possible hazard intensity measures that modellers choose to use, but having a list of common ones would help newcoming modellers and facilitate pairing of hazard and vulnerability modules across providers. 

Obviously a very important design question, so i welcome lots of comments and suggestions

**Comment by MattDonovan82** [comment]
@johcarter including the common hazard types and intensity measures in OED would definitely be a good start. Could the more unique/less common ones then be captured in the model settings file in parallel to the ones in OED or could this pose an issue with duplication? I only ask as a hybrid approach may accommodate more users initially.

**Comment by johcarter** [comment]
yes I think so @MattDonovan82 . as long as the meta data is well structured so that it would not matter whether an OED identifier was present or not.

**Comment by stufraser1** [comment]
GFDRR, in developing the Risk Data Library standard looked at this question, with a consortium of hazard experts. 

An example of metadata shows fields to give specific information related to hazards, added to ISO geospatial metadata - added with metadata_type RDL_DATA_SPECIFIC. 
[hazard_maps_flu_def_updated.meta.csv](https://github.com/OasisLMF/OpenDataStandards/files/9022148/hazard_maps_flu_def_updated.meta.csv)
Data with these metadata will be stored on the World Bank data catalogue. 
This page of documentation shows how RDL stores hazard data, with associated data fields and values: https://docs.riskdatalibrary.org/hazard.html

Fields 'Type' and  'Process' use the hazard taxonomy defined at: https://docs.riskdatalibrary.org/taxonomy.html


We defined a long list of intensity measures per hazard process, which you can use. Requesting @MamadioCMCC to provide that list here.

**Comment by matamadio** [comment]
In GFDRR RDL, metric and unit are tied together.
The following is the list we put together, knowing it will likely be expanded:

HAZARD | UNIT | DESCRIPTION
-- | -- | --
EQ | PGA:g | Peak ground acceleration in g
EQ | PGA:m/s2 | Peak ground acceleration in m/s2 (meters per second squared)
EQ | PGV:m/s | Peak ground velocity in m/s
EQ | AvgSa:m/s2 | Average spectral acceleration
EQ | Sd(T1):m | Spectral displacement
EQ | Sv(T1):m/s | Spectral velocity
EQ | PGDf:m | Permanent ground deformation
EQ | D:s | Significant duration
EQ | IA:m/s | Arias intensity (IÎ±) or (IA) or (Ia)
EQ | Neq:- | Effective number of cycles
EQ | EMS:- | European macroseismic scale
EQ | MMI:- | Modified Mercalli Intensity
EQ | CAV:m/s | Cumulative absolute velocity
EQ | D_B:s | Bracketed duration
FL, CF | fl_wd:m | Flood water depth
FL, CF | fl_wv:m/s | Flood flow velocity
WI | v_ect(3s):km/h | 3-sec at 10m sustained wind speed (kph)
WI | v_ect(1m):km/h | 1-min at 10m sustained wind speed (kph)
WI | v_etc(10m):km/h | 10-min sustained wind speed (kph)
WI | PGWS_tcy:km/h | Peak gust wind speed
LS | ls_fd:m | Landslide flow depth
LS | I_DF:m3/s2 | Debris-flow intensity index
LS | v_lsl:m/s2 | Landslide flow velocity
LS | ls_mfd:m | Maximum foundation displacement
LS | SD_lsl:m | Landslide displacement
TS | Rh_tsi:m | Tsunami wave runup height
TS | d_tsi:m | Tsunami inundation depth
TS | MMF:m4/s2 | Modified momentum flux
TS | F_drag:kN | Drag force
TS | Fr:- | Froude number
TS | v_tsi:m/s | Tsunami velocity
TS | F_QS:kN | Quasi-steady force
TS | MF:m3/s2 | Momentum flux
TS | h_tsi:m | Tsunami wave height
TS | Fh_tsi:m | Tsunami Horizontal Force
VO | h_vaf:m | Ash fall thickness
VO | L_vaf:kg/m2 | Ash loading
DR | CMI:- | Crop Moisture Index
DR | PDSI:- | Palmer Drought Severity Index
DR | SPI:- | Standard Precipitation Index

**Comment by johcarter** [comment]
This looks ideal, thank you @stufraser1 @matamadio ! 
At first glance, it covers most of the common hazards in catastrophe modelling. The ones I can't see are hail, wildfire. fire following earthquake, sprinkler damage.

Separately I wonder to what extent we could/should bring ORD model meta data in line with RDL. Many of the attributes in the example hazard map are also appropriate for catastrophe models.

**Comment by stufraser1** [comment]
I think we'd be happy to extend the RDL code list to include hail, wildfire, fire following earthquake, sprinkler damage
What intensity and units do you use for these?

We can share the metadata RDL uses for loss estimates too.
Here is the [data structure](https://docs.riskdatalibrary.org/loss.html) for loss datasets. 
@matamadio do you have a metadata file example too?

Would be great to align them, and a good outcome since GFDRR and Oasis had a joint workshop last year discussing how we can better align / support each other. 

@pchrzanowski (now leading RDL) and @dickiewhit will be interested in this development.

**Comment by johcarter** [comment]
Thanks @stufraser1. I think hail stone diameter in cm is common. I'm not sure about fire intensity and sprinkler damage, but we could ask the oasis modeller community for input on this.

happy to continue the conversation on data standard alignment, I'll review the documents you link to.

**Comment by matamadio** [comment]
The [metadata file (json)](https://github.com/GFDRR/rdl-docs/blob/main/docs/schema/rdl_schema_0.1.json) is still a work in progress.

Some comments on the proposed additional metrics:
- Wildfire is already considered by the hazard taxonomy, but we are missing metrics which should reflect the standard layer in use. The one we have uses Fire Weather Index for hazard intensity; additional metrics would be used in relation to different datasets i.e. modis burnt land or max fire temperature maps.
- Hail is not currently listed in our hazard taxonomy. We did consider to include it as meteorological hazard, but we couldn't find extensitve and reliable hazard maps for that.
- Fire following earthquake: can it be classified as wildfire with seismic trigger? And could it use a Wildfire metric?
- Sprinkler damage: this is an impact metric rather than hazard, as such it should be classified by the exposure schema.

**Comment by stufraser1** [comment]
To clarify on sprinkler damage: rather than damage _to_ sprinklers, industry models can include loss incurred due to the additional damage caused _by_ the water sprayed from sprinklers in a fire. 
In that sense it is coded as a peril in OED or AIR CEDE datasets, for locations/buildings which have that included in its insurance coverage.  This effect is modelled as a factor (in my recollection) which increases the loss incurred from the main hazard. Its something the development sector would be modelling very rarely, if at all. Given RDL is designed for development sector use, it may be something we choose not to support (per the decisions on simplifying GED4GEM exposure taxonomy when developing the simplified but hazard-expanded GED4ALL).

**Comment by dgregory-clgx** [comment]
Hi,
May have missed them, but adding some more parameters here:

![image](https://user-images.githubusercontent.com/42841277/177841699-0a67cc9e-f37c-4b38-b530-3ff9ad346274.png)

For Fire Following & Sprinkler Leakage, often these sub-perils include a 'trigger' component, as well as severity (typically captured by the 'hazard') component.  We wouldn't look to support hazard parameter outputs for these sub-perils at this time.

**Comment by MattDonovan82** [comment]
Hi @johcarter bringing this back on the radar as the OED/RDL works picks up again.

**Comment by aiste-kalinauskaite** [comment]
@johcarter for the missing units from some perils (wildfire, etc) is best to get in touch with the model vendors who have models for them and then include what they are using.
Great idea to standardise the units!

**Comment by stufraser1** [comment]
Rather than using the list pasted above in this issue, please use the latest dev codelist from RDLS:
https://github.com/GFDRR/rdl-standard/blob/0.3-codelist-updates/schema/codelists/open/IMT.csv?plain=1

And the production version:
https://github.com/GFDRR/rdl-standard/blob/main/schema/codelists/open/IMT.csv?plain=1

**Comment by aiste-kalinauskaite** [comment]
@stufraser1 is there a way to associate all the units with the respective perils they cover? In some cases it's pretty obvious, but the further down the list you go, the more obscure they get.

**Comment by matamadio** [comment]
> @stufraser1 is there a way to associate all the units with the respective perils they cover? In some cases it's pretty obvious, but the further down the list you go, the more obscure they get.

The new schema release and related [metadata editor](https://gfdrr.github.io/CCDR-tools/_static/RDL_MDE.html) provide auto-mapping of hazards to their relevant metrics and units. Please note this is still a draft, work in progress.
